src/ArmTemplates/Commands/Abstractions/ICommand.cs

```diff
@@ -0,0 +1,12 @@
+﻿// --------------------------------------------------------------------------
+//  <copyright file="ICommandLineApplication.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// --------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Abstractions
+{
+    interface ICommand
```

Comment:
This is an antipattern
- https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1040
- https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/interface

Reply:
Agree, thanks. Just to clarify my intentions - I am not familiar with `McMaster.CommandLine` library to work with parsing arguments and was pretty hard to do it with that library (at least for me)

I decided to rework this part of program - migrated to `CommandLineParser`, made all the input configurations really stable with attribute namings for parameters (not in-code hidden somewhere).

So now with dependency injection at least for me Program.Main seems like a very easy to read. Hope for you too